### PR TITLE
Web console: add a button to get out of restricted mode, make capability detection more robust

### PR DESCRIPTION
**Pull request by vogievetsky**
Making the capability detection better and letting the user override it if needed.

image

### CODE REVIEW

File: web-console/src/utils/capabilities.ts [hunk 1]
```diff
@@ -55,7 +56,7 @@ export class Capabilities {
     // Check SQL endpoint
     try {
       await Api.instance.post(
-        '/druid/v2/sql',
+        '/druid/v2/sql?capabilities',
```
Comment (clintropolis) [inline web-console/src/utils/capabilities.ts:59]: is the `?capabilities` query parameter just for fun?

**Review by clintropolis — APPROVED** [review]
i guess the button is to avoid a page refresh if it fails to detect capabilities correctly?

**Comment by vogievetsky** [comment]
> i guess the button is to avoid a page refresh if it fails to detect capabilities correctly?

Or if the capabilities detection code fails in general for some reason and you want to override it.

File: web-console/src/utils/capabilities.ts (hunk 1 above)
Comment (vogievetsky) [reply web-console/src/utils/capabilities.ts:59]: it is mostly for being able to identify these bad boys in the Network tab:

image

**Pull request merged**